src/api/hooks.ts

```diff
@@ -110,43 +112,38 @@ export function useWhoBlockedMe(websiteName: string) {
   return { loading, unblocked, blocked, possible, unknown, error };
 }
 export function useRanking(countryName: string) {
+  const apiClient = useApiClient();
   const [loading, setLoading] = useState(true);
-  const [data, setData] = useState<CountryRankingWBlocks>();
+  const [data, setData] = useState<CountryScoreWBlocked>();
+  const [blockedMap, setBlockedMap] = useState<Map<string, WebsiteStatus>>();
   const [error, setError] = useState("");
+
   const fetchData = async (countryName: string) => {
-    axiosRetry(axios, {
-      retries: 3,
-      retryDelay: (retryCount) => {
-        return Math.random() * (Math.min(10000, 1000 * (retryCount ** 2))); // https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/
-      }
-    });
-    const uri = new URL(`v1/countries/rankings/${countryName}`, API_URI);
     try {
-      const request = await axios.get(uri.href);
-      const resp = request.data as CountryRankingWBlocks;
+      const resp = (await apiClient.rankings(countryName)).result;
       const blockedMap = new Map<string, WebsiteStatus>();
-      for (const web of resp.Websites) {
+      for (const web of resp.websites ?? []) {
         let blocked: WebsiteStatus = { Blocked: false };
-        for (const bWeb of resp.BlockedWebsites) {
+        for (const bWeb of resp.blockedWebsites ?? []) {
```

Comment:
what does double ? mean in typescript

Reply:
Nullish Coalescing, aka if this is null/undefined use `[]`